models/schemas/levelSchema.ts

```diff
@@ -200,78 +209,122 @@ async function calcStats(lvl: Level) {
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export async function calcPlayAttempts(levelId: Types.ObjectId, options: any = {}) {
-  const countJustBeaten = await PlayAttemptModel.countDocuments({
-    levelId: levelId,
-    attemptContext: AttemptContext.JUST_BEATEN,
-  });
-
   // sumDuration is all of the sum(endTime-startTime) within the playAttempts
-  const sumDuration = await PlayAttemptModel.aggregate([
-    {
-      $match: {
-        levelId: levelId,
-        attemptContext: { $ne: AttemptContext.BEATEN },
+  const [countJustBeaten, sumDuration, sumDurationP95, uniqueUsersList] = await Promise.all([
+    PlayAttemptModel.countDocuments({
+      levelId: levelId,
+      attemptContext: AttemptContext.JUST_BEATEN,
+    }),
+    PlayAttemptModel.aggregate([
+      {
+        $match: {
+          levelId: levelId,
+          attemptContext: { $ne: AttemptContext.BEATEN },
+        }
+      },
+      {
+        $group: {
+          _id: null,
+          sumDuration: {
+            $sum: {
+              $subtract: ['$endTime', '$startTime']
+            }
+          }
+        }
       }
-    },
-    {
-      $group: {
-        _id: null,
-        sumDuration: {
-          $sum: {
+    ], options),
+    PlayAttemptModel.aggregate([
+      {
+        $match: {
+          levelId: levelId,
+          attemptContext: { $ne: AttemptContext.BEATEN }
+        }
+      },
+      {
+        $project: {
+          duration: {
             $subtract: ['$endTime', '$startTime']
           }
         }
+      },
+      { $sort: { duration: 1 } },
+      {
+        $group: {
+          _id: null,
+          durations: { $push: '$duration' },
+          count: { $sum: 1 }
+        }
+      },
+      {
```

Comment:
Can we combine `sumDuration` and `sumDurationP95`? Should have all the data for `sumDuration` at this point

Reply:
yeah good idea

Reply:
actually i realized when trying to do this that this is risky and not actually as straightforward since i do a bunch of groups. probably safer for now to keep it separate